packages/react/src/components/Pagination/index.tsx

```diff
@@ -81,16 +81,20 @@ export const Pagination = ({
     onPageChange(Math.min(page + 1, pagesCount));
   };
 
+  useEffect(() => {
+    setGotoValue(page + "");
+  }, [page]);
+
   const gotoPage = () => {
     let value = +gotoValue;
-    if (value < 0) {
+    if (value < 0 || !value) {
       value = 1;
     }
     if (value > pagesCount) {
       value = pagesCount;
     }
     onPageChange(value);
-    setGotoValue("");
+    setGotoValue(value + "");
```

Comment:
I feel like we do twice the same thing.
`onPageChange` will change `page` and trigger the `useEffect` that execute `setGotoValue`

Reply:
good catch